Source/WebKit/WPEPlatform/wpe/WPECursorTheme.cpp

```diff
@@ -73,42 +73,54 @@ static GUniquePtr<char> cursorsPath(const char* basePath, Vector<GUniquePtr<char
 
 WTF_MAKE_TZONE_ALLOCATED_IMPL(CursorTheme);
 
-std::unique_ptr<CursorTheme> CursorTheme::create(const char* name, uint32_t size)
+static std::unique_ptr<CursorTheme> tryCreateTheme(const char* basePath, uint32_t size)
 {
-    auto findThemePath = [](const char* name) -> GUniquePtr<char> {
-        GUniquePtr<char> path(g_build_filename(g_get_user_data_dir(), "icons", name, nullptr));
-        if (g_file_test(path.get(), G_FILE_TEST_IS_DIR))
-            return path;
-
-        path.reset(g_build_filename(g_get_home_dir(), ".icons", name, nullptr));
-        if (g_file_test(path.get(), G_FILE_TEST_IS_DIR))
-            return path;
-
-        auto* dataDirs = g_get_system_data_dirs();
-        for (unsigned i = 0; dataDirs[i]; ++i) {
-            path.reset(g_build_filename(dataDirs[i], "icons", name, nullptr));
-            if (g_file_test(path.get(), G_FILE_TEST_IS_DIR))
-                return path;
-        }
+    if (!g_file_test(basePath, G_FILE_TEST_IS_DIR))
         return nullptr;
-    };
 
-    if (auto basePath = findThemePath(name)) {
-        Vector<GUniquePtr<char>> inheritedThemes;
-        auto path = cursorsPath(basePath.get(), inheritedThemes);
-        if (path || !inheritedThemes.isEmpty()) {
-            if (!path) {
-                // If there's no cursors path, use the first inherited theme.
-                path = WTFMove(inheritedThemes[0]);
-                inheritedThemes.removeAt(0);
-            }
-
-            return makeUnique<CursorTheme>(WTFMove(path), size, WTFMove(inheritedThemes));
-        }
-        g_warning("Could not find any cursors (search started from '%s')", basePath.get());
-    } else
-        g_warning("Could not find any base paths for cursors search");
+    Vector<GUniquePtr<char>> inheritedThemes;
+    auto path = cursorsPath(basePath, inheritedThemes);
+    if (!path && inheritedThemes.isEmpty())
+        return nullptr;
+
+    if (!path) {
+        // If there's no cursors path, use the first inherited theme.
+        path = WTFMove(inheritedThemes[0]);
+        inheritedThemes.removeAt(0);
+    }
+
+    return makeUnique<CursorTheme>(WTFMove(path), size, WTFMove(inheritedThemes));
+}
+
+std::unique_ptr<CursorTheme> CursorTheme::create(const char* name, uint32_t size)
```

Comment:
I feel like the intent in this method is a loop:

1. user-data {theme} icons
2. home {theme} .icons
3. system-data {theme} icons
4. again, with "default" as theme
5. Adwaita
6. warn and end

I believe the code could be cleaner if we wrote it in that way: looping through a list of paths in priority order, and inside the body of the loop building the filename and doing `tryCreateTheme`